### ISSUE TITLE
TLS connections to RabbitMQ

### ISSUE THREAD

**Issue by CSTDev** [Templum/rabbitmq-connector#140]
**Feature request** (I don't think it's currently possible?)
All connections to the Rabbitmq I'm using are required to use TLS for both the connection and client authentication. Would it be possible to add some configuration to allow a certificate, key and CA certificate chain to be provided to the underlying rabbit client?

I've noticed the library being used for the connection has a _DialTLS_ method which I think could be used.

Happy to discuss and implement.

**Comment by Templum** [comment]
Hello @CSTDev, sounds legit. I will have a look into the `DialTLS` method to check what it expects.

**Comment by Templum** [comment]
@CSTDev I will proceed with implementing this FR, please make sure to follow the progress on the PR once I started. As I will probably need to rely on you for validation.

**Comment by Templum** [comment]
@CSTDev would you be so kind to try out the coding from #142 and see if it works for you? The README on the branch already contains the necessary documentation with respect to the TLS setup.

The only thing missing over at the PR are the tests, which I will as I have time. And a manual validation.

**Comment by Templum** [comment]
@CSTDev did you find the time to verify it ?

**Comment by CSTDev** [comment]
Code looks good, haven't had a chance to verify it yet will try to get to it the end of this week.

**Comment by CSTDev** [comment]
Any chance of adding an optional password for the client key please? Looks like passing the file to [DecryptPEMBlock ](https://golang.org/pkg/crypto/x509/#DecryptPEMBlock ) with the password, and then into X509KeyPair should do it. 

Otherwise I've got it working, thanks!

**Comment by Templum** [comment]
@CSTDev thanks for getting back and testing. I'm concerned about the deprecation warning and would like to not support something already marked deprecated 

```
Deprecated: Legacy PEM encryption as specified in RFC 1423 is insecure by design. Since it does not authenticate the ciphertext, it is vulnerable to padding oracle attacks that can let an attacker recover the plaintext.
```

**Comment by CSTDev** [comment]
That makes sense, I somehow missed that when I read it!

**Comment by Templum** [comment]
@CSTDev alright, then I believe I will merge the PR in the next days. And release a new version. Also thanks for your good catch comment over at the PR, I resolved it today along with some flood of new tests.

**Comment by Templum** [comment]
After some issues with the pipeline, due to a failure in one of the GitHub actions. I finally released this feature in [v1.0.3](https://github.com/Templum/rabbitmq-connector/releases/tag/v1.0.3)
